libraries/shared/storage/decoder.go

```diff
@@ -45,9 +45,13 @@ func Decode(diff types.PersistedDiff, metadata types.ValueMetadata) interface{}
 	case types.Bytes32:
 		return diff.StorageValue.Hex()
 	case types.PackedSlot:
-		return decodePackedSlot(diff.StorageValue.Bytes(), metadata.PackedTypes)
+		decodedSlotData, err := decodePackedSlot(diff.StorageValue.Bytes(), metadata.PackedTypes)
+		if err != nil {
+			return err
```

Comment:
I think we might want to change this function signature to return `(interface{}, error)` instead of just returning the error as an interface, because otherwise I think we might trigger a panic downstream when attempting to cast an interface to a string of bytes or whatever when it's actually an error